config/retry.go

```diff
@@ -23,12 +26,18 @@ type RetryFunc func(int) (bool, time.Duration)
 type RetryConfig struct {
 	// Attempts is the total number of maximum attempts to retry before letting
 	// the error fall through.
+        // 0 means unlimited.
 	Attempts *int
 
 	// Backoff is the base of the exponentialbackoff. This number will be
 	// multipled by the next power of 2 on each iteration.
 	Backoff *time.Duration
 
+        // MaxBackoff is an upper limit the result of the exponential backoff
+        // calculation
+        // 0 means unlimited.
```

Comment:
I would inline these with the previous line. I would also say `A MaxBackoff of zero means there is no limit to the exponential growth of the backoff.`